drivers/mspi/mspi_nrfe.c

```diff
@@ -0,0 +1,718 @@
+		if (cfg->dqs_enable) {
+			LOG_ERR("DQS signal is not supported.");
+			return -ENOTSUP;
+		}
+	}
+
+	memcpy((void *)&drv_data->dev_cfg, (void *)cfg, sizeof(drv_data->dev_cfg));
+	drv_data->dev_id = *dev_id;
+
+	return send_config(NRFE_MSPI_CONFIG_DEV, (void *)cfg, sizeof(struct mspi_dev_cfg));
+}
+
+static int api_get_channel_status(const struct device *dev, uint8_t ch)
+{
+	return 0;
+}
+
+/**
+ * @brief Send a transfer packet to the eMSPI controller.
+ *
+ * @param dev eMSPI controller device
+ * @param packet Transfer packet containing the data to be transferred
+ * @param timeout Timeout in milliseconds
+ *
+ * @retval 0 on success
+ * @retval -ENOTSUP if the packet is not supported
+ * @retval -ENOMEM if there is no space in the buffer
+ * @retval -ETIMEDOUT if the transfer timed out
+ */
+static int xfer_packet(struct mspi_xfer_packet *packet, uint32_t timeout)
+{
+	int rc;
+	uint32_t struct_size = sizeof(struct mspi_xfer_packet);
+	uint32_t len = struct_size + packet->num_bytes + 1;
+	uint8_t buffer[len];
+	enum nrfe_mspi_opcode opcode = (packet->dir == MSPI_RX) ? NRFE_MSPI_TXRX : NRFE_MSPI_TX;
+
+	buffer[0] = (uint8_t)opcode;
+	memcpy((void *)&buffer[1], (void *)packet, struct_size);
+	memcpy((void *)(&buffer[1] + struct_size), (void *)packet->data_buf, packet->num_bytes);
+
+	rc = mspi_ipc_data_send(opcode, buffer, len);
+	if (rc < 0) {
+		LOG_ERR("%d: Packet transfer error: %d", __LINE__, rc);
+	}
+
+	rc = nrfe_mspi_wait_for_response(opcode, timeout);
+	if (rc < 0) {
+		LOG_ERR("%d: FLPR Xfer response timeout: %d", __LINE__, rc);
+		return rc;
+	}
+
+	/* Wait for the transfer to complete and receive data. */
+	if (packet->dir == MSPI_RX) {
+		if ((ipc_receive_buffer != NULL) && (ipc_received > 0)) {
+			memcpy((void *)packet->data_buf, (void *)ipc_receive_buffer, ipc_received);
+			packet->num_bytes = ipc_received;
+
+			/* Clear the receive buffer pointer and size */
+			ipc_receive_buffer = NULL;
+			ipc_received = 0;
+		}
+	}
+
+	return rc;
+}
+
+/**
+ * @brief Initiates the transfer of the next packet in an MSPI transaction.
+ *
+ * This function prepares and starts the transmission of the next packet
+ * specified in the MSPI transfer configuration. It checks if the packet
+ * size is within the allowable limits before initiating the transfer.
+ *
+ * @param xfer Pointer to the mspi_xfer structure.
+ * @param packets_done Number of packets that have already been processed.
+ *
+ * @retval 0 If the packet transfer is successfully started.
+ * @retval -EINVAL If the packet size exceeds the maximum transmission size.
+ */
+static int start_next_packet(struct mspi_xfer *xfer, uint32_t packets_done)
+{
+	struct mspi_xfer_packet *packet = (struct mspi_xfer_packet *)&xfer->packets[packets_done];
+
+	if (packet->num_bytes >= MAX_TX_MSG_SIZE) {
+		LOG_ERR("Packet size to large: %u. Increase SRAM data region.", packet->num_bytes);
+		return -EINVAL;
+	}
+
+	return xfer_packet(packet, xfer->timeout);
```

Comment:
If this is the only place `xfer_packet` function is used, then maybe `xfer_packet` should be deleted and its body pasted here?